src/replication.rs

```diff
@@ -94,7 +150,7 @@ async fn replicate_public_key(inner: &ReplicationState, public_key: &String, url
 
         let stream = stream::iter(packets);
 
-        client.put(tonic::Request::new(stream)).await?;
+        cached_client.lock().await.put(tonic::Request::new(stream)).await?;
```

Comment:
Should we do an exponential backoff on the put call as well, or just on the connect? Does the connect call actually initiate the connection, or just create a stub for the given endpoint?